packages/eslint-plugin/src/rules/consistent-callback-type/index.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+const { AST_NODE_TYPES } = require('@typescript-eslint/experimental-utils');
+const createRule = require('../../utils/createRule');
+
+module.exports = createRule({
+  name: 'consistent-callback-type',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Enforce callback props to be typed with `EventHandler`',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      invalidType: 'callback props should be typed with EventHandler<T>',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      'TSTypeAliasDeclaration[id.name=/.*Props$/] TSTypeLiteral': function (node) {
+        // @ts-ignore
+        node.members.forEach(member => {
+          if (
+            member.type === 'TSPropertySignature' &&
```

Comment:
lets avoid magic strings and only uses officially provided node types

```suggestion
            member.type === AST_NODE_TYPES.TSPropertySignature &&
```